### PR TITLE
Feat/add status condition tests #94

### LINKED ISSUE #94: 📝 [FEAT] Add Unit Tests for StatusCondition Enum and Conversions
### 🔍 Scope
The `StatusCondition` enum defines various debuff or ailment states that can affect a player or beast in the combat game. It includes conversions between `StatusCondition` and both `felt252` and `u8`. However, there are currently no unit tests verifying these conversions.

This issue focuses on writing comprehensive tests to ensure that all enum conversions behave as expected and handle edge cases safely.

### 🔄 How to Implement the Issue

- **Step 1:** Write tests for each of the following:
  - `StatusCondition` to `felt252` using `Into<StatusCondition, felt252>`
  - `StatusCondition` to `u8` using `Into<StatusCondition, u8>`
  - `u8` to `StatusCondition` using `Into<u8, StatusCondition>`

- **Step 2:** Validate:
  - Each enum maps to the correct numeric value.
  - All valid `u8` values map correctly back to `StatusCondition`.
  - Invalid values return `StatusCondition::None` as a fallback.

- **Step 3:** Include edge cases such as inputs `255`, `100`, etc.

### ✅ Acceptance Criteria
- [ ] Unit tests exist for each direction of enum conversion.
- [ ] All conversions behave correctly for valid and invalid values.
- [ ] `StatusCondition::None` is returned safely for invalid `u8` input.
- [ ] Tests follow Cairo best practices and are clearly named.

### 📸 Additional Context
These status effects are fundamental to gameplay and must behave predictably. Testing ensures that state transitions and game logic aren't corrupted by incorrect enum behavior.

Enum source:  
https://github.com/AkatsukiLabs/DojoByExample/blob/main/backend/dojo_examples/combat_game/src/types/status_condition.cairo

### Contribution Guidelines
- Please ensure you read and follow the contribution guidelines in the project's [CONTRIBUTIONS GUIDELINES](https://github.com/AkatsukiLabs/DojoByExample/issues/3)

### PR DESCRIPTION
**Pull request by morelucks**
# Pull Request Overview

## 📝 Summary
Provide a brief overview of what this PR accomplishes.

### Related Issues
- Closes #94 

### Type of Change
Mark with an `x` all the checkboxes that apply (like `[x]`).
- [ ] 📝 Documentation (updates to README, docs, or comments)
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 👌 Enhancement (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

### What's Changed
Provide a clear and concise description of what you changed and why.

### Implementation Details
Describe how you implemented the changes.

### Technical Notes
Include any technical details that reviewers should be aware of.

## 🔧 Tests Results
Describe the tests you performed to verify your changes.

### Test Coverage
- [x] Unit Tests
- [ ] Integration Tests
- [ ] Manual Testing

### Evidence
Provide relevant evidence of testing (screenshots, test outputs, etc.).

### Testing Notes
Include any special testing considerations or edge cases checked.

## 🔜 Next Steps
Indicate actions or improvements to be taken after this PR, if applicable.

### CODE REVIEW

File: backend/dojo_examples/combat_game/src/tests/status_condition_tests.cairo [hunk 1]
```diff
@@ -0,0 +1,63 @@
+use combat_game::types::status_condition::{
+    StatusCondition,
+}; // Import your enum from the correct path
+
+#[cfg(test)]
+mod status_condition_tests {
+    use super::*; // Import everything from the parent module (combat_game)
```
Comment (jimenezz22) [inline backend/dojo_examples/combat_game/src/tests/status_condition_tests.cairo:7]: It is not good practice to use everything, we must always specify which properties we are going to use.
```suggestion
    use super:: {StatusCondition, IntoStatusConditionFelt252, IntoStatusConditionU8, IntoU8StatusCondition};
```

File: backend/dojo_examples/combat_game/src/tests/status_condition_tests.cairo [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    // Tests conversion from `StatusCondition` variants to `u8`.
+    // Ensures each variant correctly maps to its corresponding numeric representation.
+
+    #[test]
+    fn test_status_condition_to_u8() {
+        assert_eq!(StatusCondition::None.into(), 0_u8);
```
Comment (jimenezz22) [inline backend/dojo_examples/combat_game/src/tests/status_condition_tests.cairo:14]: The assertions lack descriptive messages for when tests fail. Also use variable declarations to make the tests more readable.

File: backend/dojo_examples/combat_game/src/tests/status_condition_tests.cairo [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        assert_eq!(StatusCondition::Poisoned.into(), 1_u8);
+        assert_eq!(StatusCondition::Paralyzed.into(), 2_u8);
+        assert_eq!(StatusCondition::Asleep.into(), 3_u8);
+        assert_eq!(StatusCondition::Confused.into(), 4_u8);
+        assert_eq!(StatusCondition::Burned.into(), 5_u8);
+        assert_eq!(StatusCondition::Frozen.into(), 6_u8);
+        assert_eq!(StatusCondition::Cursed.into(), 7_u8);
+    }
+
```
Comment (jimenezz22) [inline backend/dojo_examples/combat_game/src/tests/status_condition_tests.cairo:23]: Adapt the test to follow this format:
```suggestion
    #[test]
    #[available_gas(1000000)]
    fn test_status_condition_to_u8() {
        let none = StatusCondition::None;
        let poisoned = StatusCondition::Poisoned;
        
        assert_eq!(none.into(), 0_u8, "StatusCondition::None should convert to 0");
        assert_eq!(poisoned.into(), 1_u8, "StatusCondition::Poisoned should convert to 1");
}        
        
fn test_status_condition_to_felt252() {
        let none = StatusCondition::None;
        let poisoned = StatusCondition::Poisoned;
        
        assert_eq!(none.into(), 0_felt252, "StatusCondition::None should convert to 0");
        assert_eq!(poisoned.into(), 1_felt252, "StatusCondition::Poisoned should convert to 1");
}

fn test_u8_to_status_condition_valid() {
        let zero = 0_u8;
        let one = 1_u8;
        
        assert_eq!(zero.into(), StatusCondition::None, "0 should convert to StatusCondition::None");
        assert_eq!(one.into(), StatusCondition::Poisoned, "1 should convert to StatusCondition::Poisoned");
}
        
fn test_u8_to_status_condition_invalid() {
        let eight = 8_u8;
        let hundred = 100_u8;
        
        assert_eq!(eight.into(), StatusCondition::None, "8 should default to StatusCondition::None");
        assert_eq!(hundred.into(), StatusCondition::None, "100 should default to StatusCondition::None");
}
```

**Comment by jimenezz22** [comment]
@morelucks 
You made many unrequested changes to files that don't cover the scope of the issue. Remove all those changes from those files and focus on the scope of the tests for the requested enum. Finally, resolve the GitHub conflicts.

File: backend/dojo_examples/combat_game/src/constants.cairo
[the remark this answers is not in the document]
Comment (morelucks) [reply backend/dojo_examples/combat_game/src/constants.cairo:1]: I only formatted the code with 'scarb fmt'  without changing any statement

File: backend/dojo_examples/combat_game/src/models/bag.cairo
[the remark this answers is not in the document]
Comment (morelucks) [reply backend/dojo_examples/combat_game/src/models/bag.cairo:1]: I only formatted the code with 'scarb fmt' without changing any statement

File: backend/dojo_examples/combat_game/src/models/potion.cairo
[the remark this answers is not in the document]
Comment (morelucks) [reply backend/dojo_examples/combat_game/src/models/potion.cairo:1]: I only formatted the code with 'scarb fmt' without changing any statement

File: backend/dojo_examples/combat_game/src/tests/status_condition_tests.cairo (hunk 1 above)
Comment (morelucks) [reply backend/dojo_examples/combat_game/src/tests/status_condition_tests.cairo:7]: I will amend this

File: backend/dojo_examples/combat_game/src/models/bag.cairo
[the remark this answers is not in the document]
Comment (jimenezz22) [reply backend/dojo_examples/combat_game/src/models/bag.cairo:1]: > I only formatted the code with 'scarb fmt' without changing any statement

yeah but in some case, for example in this file scrab fmt reduce a couple lines of code into one sentence that can affect easy code readability

**Comment by jimenezz22** [comment]
@morelucks could you send a ss of the tests?

**Comment by morelucks** [comment]
@jimenezz22 

✅ All new tests pass.
✅ No existing tests broke.

image

**Review by jimenezz22 — APPROVED** [review]
